src/functional/iterator/embedded.py

```diff
@@ -194,16 +194,25 @@ def __array__(self, dtype: Optional[npt.DTypeLike] = None) -> np.ndarray:
         return self.data.astype(dtype, copy=False)
 
     def __array_ufunc__(self, ufunc, method, *inputs, **kwargs) -> Column:
+        # note:
+        # - we allow scalars to silently pass through and be handled correctly by numpy
+        # - we let numpy do the checking of compatible shapes
         assert method == "__call__"
-        assert all(inp.kstart == self.kstart for inp in inputs)
-        assert all(inp.data.shape == self.data.shape for inp in inputs)
-        return self.__class__(self.kstart, ufunc(*(inp.data for inp in inputs), **kwargs))
+        if not all(inp.kstart == self.kstart for inp in inputs if isinstance(inp, Column)):
+            raise ValueError("Incompatible Column.kstart")
```

Comment:
Not even sure if it helps, but in case you want to report more information in the exception you could do something like:
```suggestion
        if (wrong_kstarts := (set(inp.kstart for inp in inputs if isinstance(inp, Column)) - {self.kstart})):
            raise ValueError("Incompatible Column.kstart: it should be '{self.kstart}' but found other values: {wrong_kstarts}")
```